### PR TITLE
feat(application/ui-components): Creating UI-Components to hold application components

### PR DESCRIPTION
**Pull request by siggakogk**
# UI-Components to hold application components

## What

* Creating UI-Components
* Moving application Table from shared to UI-Components
* Using Table from Island-UI instead of custom styles

## Why

* Need a structure for components in the application system that is not form field
* Refactor Table to use island-ui for consistency

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/island-ui/core/src/lib/Table/Table.treat.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { style } from 'treat'
+import { style, globalStyle } from 'treat'
```
Comment (jeremybarbet) [inline libs/island-ui/core/src/lib/Table/Table.treat.ts:1]: Import not used?

File: libs/application/ui-components/src/components/Table.tsx [hunk 2]
```diff
@@ -106,28 +105,30 @@ export const Table = <T extends object>(
...
             ))}
-          </tr>
+          </T.Row>
         ))}
-      </thead>
+      </T.Head>
       {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
       {/* @ts-ignore - Same as above */}
-      <tbody {...getTableBodyProps()}>
+      <T.Body {...getTableBodyProps()}>
```
Comment (jeremybarbet) [inline libs/application/ui-components/src/components/Table.tsx:126]: I believe all these ts-ignore are not useful anymore?

File: libs/application/ui-components/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# UI Components
+
+This library contains reusable UI components for the application system that are not form fields.
+
+## Technology
```
Comment (jeremybarbet) [inline libs/application/ui-components/README.md:5]: It doesn't use react-hook-form for the table, is it?

**Review by jeremybarbet — COMMENTED** [review]
Very minor comments, looks great otherwise 👍

**Review by baering — APPROVED** [review]
LGTM, thanks! 👍

**Pull request merged**